packages/2d/src/components/CodeBlock.ts

```diff
@@ -99,6 +99,40 @@ export class CodeBlock extends Shape {
   private selectionProgress = createSignal<number | null>(null);
   private oldSelection: CodeRange[] | null = null;
   private diffed: MorphToken[] | null = null;
+  private currentLineCount = 0;
+  private newLineCount = 0;
+
+  protected getLineCountOfTokenArray(tokens: Token[]): number {
+    let count = 0;
+
+    for (const token of tokens) {
+      for (let i = 0; i < token.code.length; i++) {
+        if (token.code[i] === '\n') {
+          count++;
+        }
+      }
+    }
+
+    if (tokens.length > 0) {
+      count++;
+    }
+
+    return count;
+  }
+
+  @computed()
+  public getLineCount(): number {
+    const progress = this.codeProgress();
+    if (progress !== null) {
+      // Interpolate between the two line counts based on progress (lerping)
+      return Math.round(
+        this.currentLineCount +
+          progress * (this.newLineCount - this.currentLineCount),
+      );
```

Comment:
We have a function for linear interpolation:
```suggestion
      return Math.round(
        map(this.currentLineCount, this.newLineCount, progress)
      );
```